### ISSUE TITLE
Include a "all-pages" parameter

### ISSUE THREAD

**Issue by Aubreymcfato** [MusikAnimal/pageviews#79]
Hi, I was wondering if it would be possible to know the # of pageviews for "all-pages" in a range of dates. For many smaller Wikimedia projects this could be kinda easy, although I understand it's maybe not the case for bigger Wikipedias.

**Comment by MusikAnimal** [comment]
This is possible! See #68

I want to create a new tool entirely called "Siteviews" that lets you compare the pageviews of "all pages" of multiple projects.

However it would still be nice to add something to Pageviews Analysis. Maybe down under the "Totals" I could have it report the total for all pages within the given date range, in addition to whatever pages they've entered. I'll look into it! Thanks

**Comment by Aubreymcfato** [comment]
Thank you. For smaller wikis, this could be very useful, because at least we could see if some "campaigns" or tools are effective or not.

**Comment by Aubreymcfato** [comment]
(also, not related) Make sure to go in all Special:Statistics and replace stats.grok.se links :-)

**Comment by MusikAnimal** [comment]
Awesome, thanks for illustrating the use-case :) I can definitely get to the "Siteviews" at some point, basically it's just using much of the same code that Pageviews uses. Right now I'm working on #42 so you probably won't see Siteviews for another few weeks. Will reach out to you once deployed :)

**Comment by MusikAnimal** [comment]
@Aubreymcfato Siteviews is live! http://tools.wmflabs.org/siteviews

I'm going to work to improve the load time. Basically the Select2 library doesn't like that they are 800+ wikis to choose from. I'll have to implement some sort of hack to make it place nice.

Additionally I want to add an option to compare projects on an hourly level. E.g. what time of day is enwiki popular versus dewiki? I have a pretty good guess, but still might be interesting to see it plotted out.

Any other feedback welcomed. I'm still going to add a "all pages" statistic in Pageviews, that will link to the new Siteviews tool. So leaving this issue open for now.

Cheers

**Comment by Aubreymcfato** [comment]
This is great! 
I think you should promote this news on FB and/or wikimedia-l. 
I'll do it on wikisource-l.

**Comment by Aubreymcfato** [comment]
Just a question: is there a way to visualize the views divided per platform? Like a pie-chart in which you see how many visits from desktop, mobile, etc.

**Comment by MusikAnimal** [comment]
No, there is currently no way to visualize data on a per-platform basis, but that's a great idea. The issue is the tool is hardwired around the concept of seeing multiple projects, so we'd need some UI changes. Happy to look into it!

**Comment by Aubreymcfato** [comment]
Dear @MusikAnimal, another question (don't know if fits here): is it possible to aggregate the views *per project*? It would be cool (and probably useful) to know how much Wikipedia is bigger than Commons or Wikiquote... Right now it's possible fro single languages, but there are no grand totals. Thanks anyway.

**Comment by MusikAnimal** [comment]
@Aubreymcfato That's a fine idea, but a fair amount of work. There's no native way to do this with the API; we'd have to create something like [Massviews](https://tools.wmflabs.org/massviews) except for per-project data that is used on Siteviews. Totally doable, but it's expensive; Wikipedia has 300+ languages, so 300+ individual API queries. For Commons at least, you can easily ascertain that it is far less popular than Wikipedia as a whole ([example](http://tools.wmflabs.org/siteviews/?platform=all-access&source=pageviews&agent=user&range=latest-20&sites=fr.wikipedia.org|de.wikipedia.org|commons.wikimedia.org|en.wikipedia.org)). Comparing other projects may be more meaningful, but I suspect you could still deduce this through simpler comparisons. Sorry!

**Comment by Aubreymcfato** [comment]
@MusikAnimal I understand. I asked Dario if it's possibile to do something like this in (https://reportcard.wmflabs.org/#)[reportcards], which are already aggregated.

**Comment by parasteh** [comment]
Is it possible to get more than 1000 result to topviews ?

**Comment by MusikAnimal** [comment]
@parasteh It is not :( The data is supplied by the [https://wikimedia.org/api/rest_v1/?doc](RESTBase API) which only gives you the top 1,000 pages. Sorry!
